src/grid.h

```diff
@@ -58,17 +68,13 @@ struct Grid {
                 for (bool _f = true; _f;)   \
                     for (Cell *&c = g->C(x, y); _f; _f = false)
 
-    Grid(int _xs, int _ys, Cell *_c = nullptr)
-        : xs(_xs), ys(_ys), cell(_c), cells(new Cell *[_xs * _ys]) {
+    Grid(int _xs, int _ys, Cell *_c = nullptr) : xs(_xs), ys(_ys), cell(_c), cells(_xs * _ys) {
         foreachcell(c) c = nullptr;

```

Comment:
doesn't the default constructor for elements of a `vector<Cell *>` initialize elements to null? If so, don't need this extra loop

Reply:
You are perfectly right!